### PR TITLE
Add a utility script to scan Azure subscription for VM sizes and quota needed by LISA

### PR DESCRIPTION
**Pull request by lesscodingmorehappiness**
…ipeline to scan subscription and record unavailabe vm or insufficient vm quota

Tested in cloned pipeline

### CODE REVIEW

File: Utilities/ScanSubscription.ps1 [hunk 1]
```diff
@@ -0,0 +1,362 @@
+# Description: This script is used for chercking enabled VM sizes and VM core quota in the given azure subscription.
+# It will upload subscription id and unavailable VM/insufficient VM core to db
+# This script will be executed by 'Penguinator-Validate-Azure-Subscription-Info' pipeline
+# This script should be under .\lisav2\Utilities directory
+
+$clientSecret = "$(CLIENT_SECRET)"
+$clientID = $env:CLIENT_ID
+$tenantID = $env:TENANT_ID
+$subID = $env:SUBSCRIPTION_ID
```
Comment (lizzha) [inline Utilities/ScanSubscription.ps1]: Can you pass the 4 variables as the parameters of this script?

File: Utilities/ScanSubscription.ps1 [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+$lisaTestCasesXMLPath = ".\lisav2\XML\TestCases"
+$lisaVMConfigsPath = ".\lisav2\XML\VMConfigurations"
+$penguinatorSecureFilePath = $env:PENGUINATOR_SECUREFILEPATH
```
Comment (lizzha) [inline Utilities/ScanSubscription.ps1]: ditto

File: Utilities/ScanSubscription.ps1 [hunk 3]
```diff
@@ -0,0 +1,362 @@
+# Description: This script is used for chercking enabled VM sizes and VM core quota in the given azure subscription.
```
Comment (LiliDeng) [inline Utilities/ScanSubscription.ps1]: checking

Comment (lesscodingmorehappiness) [reply Utilities/ScanSubscription.ps1]: changed

File: Utilities/ScanSubscription.ps1 (hunk 1 above)
Comment (lesscodingmorehappiness) [reply Utilities/ScanSubscription.ps1]: fixed

File: Utilities/ScanSubscription.ps1 (hunk 2 above)
Comment (lesscodingmorehappiness) [reply Utilities/ScanSubscription.ps1]: fixed

File: Utilities/ScanSubscription.ps1 [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+# This script will be executed by 'Penguinator-Validate-Azure-Subscription-Info' pipeline
+# This script should be under .\lisav2\Utilities directory
+
+Param
+(
+    [String] $clientSecret,
+    [String] $clientID,
+    [String] $tenantID,
+    [String] $subID,
+    [String] $lisaTestCasesXMLPath = '.\lisav2\XML\TestCases',
```
Comment (lizzha) [inline Utilities/ScanSubscription.ps1]: "lisav2" seems to be a custom folder, can we set it to be ".\xml\TestCases", and assume the current execution folder is the cloned repo folder?

File: Utilities/ScanSubscription.ps1 [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+    $finalXml += "</TestSetup>"
+    ([xml]$finalXml).Save(".\lisav2\mergedVMConfig.xml")
+    return [xml]$finalXml
+}
+
+function Measure-SubscriptionCapabilities() {
+    Write-LogInfo "Measure VM capabilities of current subscription..."
+    if (!$script:SubscriptionVMResourceSkus -or !$script:TestableLocations) {
+        $regionScopeFromUser = @()
+        $RegionAndStorageMapFile = ".\lisav2\XML\RegionAndStorageAccounts.xml"
```
Comment (lizzha) [inline Utilities/ScanSubscription.ps1]: ditto

File: Utilities/ScanSubscription.ps1 [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+Param
+(
+    [String] $clientSecret,
+    [String] $clientID,
+    [String] $tenantID,
+    [String] $subID,
+    [String] $lisaTestCasesXMLPath = '.\lisav2\XML\TestCases',
+    [String] $lisaVMConfigsPath = '.\lisav2\XML\VMConfigurations',
+    [String] $penguinatorSecureFilePath
+)
```
Comment (lizzha) [inline Utilities/ScanSubscription.ps1:17]: can we add one more parameter for parallel count, the required quota may change per parallel count.
Say the parallel count is n, then the minimal quota for a size is the sum of n cases with top n required cores.

Comment (lesscodingmorehappiness) [reply Utilities/ScanSubscription.ps1:17]: Sure. 
To clarify this:
1. what should be the default n? 2?
2. after getting the sum of top n required cores, do we still want to round up to closest 100s?

File: Utilities/ScanSubscription.ps1 (hunk 4 above)
Comment (lesscodingmorehappiness) [reply Utilities/ScanSubscription.ps1]: Currently (I mean in my test pipeline) I clone lisa to $executionFolder/lisav2 to differentiate lisa codes from other codes/files under execution folder. I prefer keep it as is.

Comment (lizzha) [reply Utilities/ScanSubscription.ps1]: you can add "pushd lisa" in your pipeline job.
If we want to keep it as a generic utility, it's better not having the customized path, or we can pass "lisa" from parameter

File: Utilities/ScanSubscription.ps1 (hunk 6 above)
Comment (lizzha) [reply Utilities/ScanSubscription.ps1:17]: currently please set n default to 4.
Yes we still need to round up to 100

File: Utilities/ScanSubscription.ps1 [hunk 7]
```diff
@@ -0,0 +1,437 @@
...
+(
+    [String] $clientSecret,
+    [String] $clientID,
+    [String] $tenantID,
+    [String] $subID,
+    [String] $lisaDir = '.',
+    [String] $lisaTestCasesXMLPath = "$lisaDir\XML\TestCases",
+    [String] $lisaVMConfigsPath = "$lisaDir\XML\VMConfigurations",
+    [String] $penguinatorSecureFilePath,
+    [int] $parrallelCount = 4
```
Comment (lizzha) [inline Utilities/ScanSubscription.ps1:16]: parallel

**Review by lizzha — APPROVED** [review]

**Pull request merged**